noir_stdlib/src/field/bn254.nr

```diff
@@ -166,6 +190,15 @@ mod tests {
         assert_gt(0 - 1, 0);
     }
 
+    #[test]
+    unconstrained fn check_assert_gt_unconstrained() {
+        assert_gt(1, 0);
+        assert_gt(0x100, 0);
+        assert_gt((0 - 1), (0 - 2));
+        assert_gt(TWO_POW_128, 0);
+        assert_gt(0 - 1, 0);
+    }
```

Comment:
This test enusres that the branch where it's unconstrained is correct, for example, if assert_gt was like this:
```rust
pub fn assert_gt(a: Field, b: Field) {
    if within_unconstrained() {
        assert(compute_lt(b, a, 16));
    } else {
```
This will pass the constraiend test and not the unconstrained

Reply:
:facepalm: yeah ofc.